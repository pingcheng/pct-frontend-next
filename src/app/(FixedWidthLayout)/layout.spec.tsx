import { render, screen } from "@testing-library/react";
import FixedWidthLayout from "./layout";
import React from "react";

// Mock the NavBar and Footer components
jest.mock("@/components/NavBar/NavBar", () => ({
  NavBar: () => <div data-testid="navbar">NavBar Component</div>,
}));

jest.mock("@/components/Footer", () => ({
  Footer: () => <div data-testid="footer">Footer Component</div>,
}));

describe("FixedWidthLayout", () => {
  test("should match snapshot", () => {
    const { container } = render(
      <FixedWidthLayout>
        <div data-testid="child-content">Test Content</div>
      </FixedWidthLayout>
    );
    expect(container).toMatchSnapshot();
  });

  test("should render NavBar component", () => {
    render(
      <FixedWidthLayout>
        <div>Test Content</div>
      </FixedWidthLayout>
    );

    expect(screen.getByTestId("navbar")).toBeInTheDocument();
  });

  test("should render Footer component", () => {
    render(
      <FixedWidthLayout>
        <div>Test Content</div>
      </FixedWidthLayout>
    );

    expect(screen.getByTestId("footer")).toBeInTheDocument();
  });

  test("should render children content", () => {
    render(
      <FixedWidthLayout>
        <div data-testid="child-content">Test Content</div>
      </FixedWidthLayout>
    );

    expect(screen.getByTestId("child-content")).toBeInTheDocument();
    expect(screen.getByTestId("child-content")).toHaveTextContent(
      "Test Content"
    );
  });

  test("should have the fixed-container class on the main content", () => {
    render(
      <FixedWidthLayout>
        <div>Test Content</div>
      </FixedWidthLayout>
    );

    const container = screen.getByText("Test Content").parentElement;
    expect(container).toHaveClass("fixed-container");
    expect(container).toHaveClass("pt-2");
  });
});
