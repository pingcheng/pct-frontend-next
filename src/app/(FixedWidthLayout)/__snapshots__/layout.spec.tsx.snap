// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FixedWidthLayout should match snapshot 1`] = `
<div>
  <div
    data-testid="navbar"
  >
    NavBar Component
  </div>
  <div
    class="page-container pt-2"
  >
    <div
      data-testid="child-content"
    >
      Test Content
    </div>
  </div>
  <div
    data-testid="footer"
  >
    Footer Component
  </div>
</div>
`;
